scalyr_agent/util.py

```diff
@@ -1316,7 +1316,12 @@ def join(self, timeout=None):
             indefinitely.
         @type timeout: float|None
         """
-        threading.Thread.join(self, timeout)
+        try:
+            threading.Thread.join(self, timeout)
+        except RuntimeError:
+            # Calling .join() on thread which is not started should not be considered fatal
```

Comment:
I guess I am missing something but I cannot find the place in this PR where this error suppression may be needed.

Reply:
It's not related directly to this PR, but I encountered this issue multiple times when testing those changes.

In short - this edge case was there for a while already.

Reply:
Here is an example:

```bash
2021-01-20 12:34:32.710Z ERROR [core] [scalyr_logging.py:618] Failed to stop the metric log due to an exception :stack_trace:
  Traceback (most recent call last):
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/monitors_manager.py", line 201, in stop_manager
      monitor.stop(join_timeout=max_wait)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 4254, in stop
      self.__container_checker.stop(wait_on_join, join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 2788, in stop
      self.__thread.stop(wait_on_join=wait_on_join, join_timeout=join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1297, in stop
      self.join(join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1319, in join
      threading.Thread.join(self, timeout)
    File "/home/kami/.pythonz/pythons/CPython-3.6.9/lib/python3.6/threading.py", line 1051, in join
      raise RuntimeError("cannot join thread before it is started")
  RuntimeError: cannot join thread before it is started
```

Technically, we could also try to re-order the code so we never call ``.join()`` on a thread which has been stopped / not started yet, but that would be a larger and more risky change.

Reply:
oh, sorry I haven't noticed your comment above. I think it's fine if we are sure that we do not also suppress any other errors, first of all - the timeout error. But I am pretty sure that there should be another exception type.